controllers/front/CartController.php

```diff
@@ -392,10 +392,21 @@ protected function processChangeProductInCart()
         }
 
         $product = new Product($this->id_product, true, $this->context->language->id);
+
+        $productAttributes = null;
+        if ($product->hasAttributes()) {
+            $separator = Configuration::get('PS_ATTRIBUTE_ANCHOR_SEPARATOR');
+            $attributes = Product::getAttributesParams($this->id_product, $this->id_product_attribute);
+            foreach ($attributes as $attribute) {
+                $productAttributes .= $attribute['group'] . ' : ' . $attribute['name'] . ' ' . $separator . ' ';
+            }
+            $productAttributes = rtrim($productAttributes, $separator . ' ');
+        }
+
         if (!$product->id || !$product->active || !$product->checkAccess($this->context->cart->id_customer)) {
             $this->{$ErrorKey}[] = $this->trans(
-                'This product (%product%) is no longer available.',
-                ['%product%' => $product->name],
+                'The product %product% is no longer available.',
+                ['%product%' => $product->name . (isset($productAttributes) ? ' ' . $productAttributes : '')],
```

Comment:
```suggestion
                ['%product%' => $product->name . (!empty($productAttributes) ? ' ' . $productAttributes : '')],
```

Reply:
I see you copy & paste this check on many lines. Maybe create a variable above to store the product name :)